libsycl/include/sycl/detail/config.hpp

```diff
@@ -0,0 +1,64 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains the declaration of the macros defining attributes for
+/// exported methods and defining API namespaces.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef __LIBSYCL_DETAIL_CONFIG_HPP
+#define __LIBSYCL_DETAIL_CONFIG_HPP
+
+#include <sycl/version.hpp>
+
+#define __LIBSYCL_BEGIN_UNVERSIONED_NAMESPACE namespace sycl {
+#define __LIBSYCL_END_UNVERSIONED_NAMESPACE }
+
+#define __LIBSYCL_BEGIN_VERSIONED_NAMESPACE                                    \
```

Comment:
Maybe name this `BEGIN_NAMESPACE_SYCL` instead? Otherwise it's not really clear what namespace you're opening.